### ISSUE TITLE
ZSTD_seekable_decompress() can hang

### ISSUE THREAD

**Issue by mlindsay** [facebook/zstd#2506]
**Describe the bug**
ZSTD_seekable_decompress() can hang with malformed input.

**To Reproduce**
Check out #2505 
```
$ cd contrib/seekable_format/tests
$ make
$ ./seekable_tests
```

**Expected behavior**
Program halts, returning an error code.

**Additional context**
This bug was found via this [fuzzer](https://fuchsia.googlesource.com/fuchsia/+/6c0911153a4b36d5a6ab86f74cc9d1a313d8e8ca/src/storage/blobfs/test/zstd-seekable-fuzzer.cc). The specific decompressor input is an input generated by the fuzzer to reproduce the issue.

**Comment by Cyan4973** [comment]
Thanks for the clear reproduction case @mlindsay.
We'll look into it and get it fix before next release.

**Comment by senhuang42** [comment]
Fixed in #2516
